--- a/src/devices/nm-device.c.orig	2019-12-17 08:59:49.000000000 +0100
+++ b/src/devices/nm-device.c	2020-01-06 17:29:08.678242168 +0100
@@ -17,7 +17,11 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 #include <linux/if_addr.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/rtnetlink.h>
 #include <linux/pkt_sched.h>
 
--- NetworkManager-1.18.4/src/nm-manager.c      2019-10-09 14:56:59.000000000 +0200
+++ NetworkManager-1.18.4-new/src/nm-manager.c  2020-03-13 15:46:39.448745439 +0100
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <asm/types.h>

 #include "nm-glib-aux/nm-c-list.h"

--- NetworkManager-1.18.4/src/platform/nm-linux-platform.c      2019-10-09 14:56:59.000000000 +0200
+++ NetworkManager-1.18.4-new/src/platform/nm-linux-platform.c  2020-03-13 15:56:40.795388006 +0100
@@ -20,6 +20,7 @@
 #include "nm-default.h"

 #include "nm-linux-platform.h"
+#include "wpan/nm-wpan-utils.h"

 #include <arpa/inet.h>
 #include <dlfcn.h>
@@ -28,7 +29,11 @@
 #include <libudev.h>
 #include <linux/fib_rules.h>
 #include <linux/ip.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/if_bridge.h>
 #include <linux/if_link.h>
 #include <linux/if_tun.h>
@@ -58,7 +63,6 @@
 #include "nm-platform-private.h"
 #include "wifi/nm-wifi-utils.h"
 #include "wifi/nm-wifi-utils-wext.h"
-#include "wpan/nm-wpan-utils.h"
 #include "nm-glib-aux/nm-io-utils.h"
 #include "nm-udev-aux/nm-udev-utils.h"

