--- a/src/devices/nm-device.c.orig	2019-12-17 08:59:49.000000000 +0100
+++ b/src/devices/nm-device.c	2020-01-06 17:29:08.678242168 +0100
@@ -17,7 +17,11 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 #include <linux/if_addr.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/rtnetlink.h>
 #include <linux/pkt_sched.h>
 
--- a/src/nm-manager.c.orig	2020-01-06 17:29:08.682242177 +0100
+++ b/src/nm-manager.c	2020-01-06 17:30:50.378482034 +0100
 #include <stdlib.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <asm/types.h>

 #include "nm-glib-aux/nm-c-list.h"
--- a/src/platform/nm-linux-platform.c.orig	2019-12-17 08:59:49.000000000 +0100
+++ b/src/platform/nm-linux-platform.c	2020-01-06 17:29:08.686242186 +0100
 #include "nm-default.h"
 
 #include "nm-linux-platform.h"
+#include "wpan/nm-wpan-utils.h"
 
 #include <arpa/inet.h>
 #include <dlfcn.h>
 #include <endian.h>
 #include <fcntl.h>
 #include <libudev.h>
 #include <linux/fib_rules.h>
 #include <linux/ip.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/if_bridge.h>
 #include <linux/if_link.h>
 #include <linux/if_tun.h>
 #include <linux/if_tunnel.h>
 #include <linux/ip6_tunnel.h>
 #include <linux/tc_act/tc_mirred.h>
 #include <netinet/icmp6.h>
 #include <netinet/in.h>
 #include <poll.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <unistd.h>
 
 #include "nm-std-aux/unaligned.h"
 
 #include "nm-utils.h"
 #include "nm-core-internal.h"
 #include "nm-setting-vlan.h"

 #include "nm-glib-aux/nm-secret-utils.h"
 #include "nm-glib-aux/nm-c-list.h"
 #include "nm-netlink.h"
 #include "nm-core-utils.h"
 #include "nmp-object.h"
 #include "nmp-netns.h"
 #include "nm-platform-utils.h"
 #include "nm-platform-private.h"
 #include "wifi/nm-wifi-utils.h"
 #include "wifi/nm-wifi-utils-wext.h"
-#include "wpan/nm-wpan-utils.h"
 #include "nm-glib-aux/nm-io-utils.h"
 #include "nm-udev-aux/nm-udev-utils.h"
